### DISCUSSION TITLE
Disconnects with new bluetooth Name in mywhoosh

### DISCUSSION THREAD

**Discussion by dadude2607** [cagnulein/qdomyos-zwift#2583] [Q&A]
Hi, I keep having disconnects in mywhoosh. Sometimes after 10 minutes (twice today), sometimes after a long time. I then have to go into the settings, disconnect the Bluetooth device and reconnect it. What is noticeable is that the virtual Bluetooth device then gets a new name (at least the numbers behind the name). How can I prevent the disconnects? Attached is a screenshot with one of the many names I get in mywhoosh. Thank you for your help

![Screenshot_20240908_204608_MyWhoosh](https://github.com/user-attachments/assets/a636c48a-1b08-4d53-8e64-f48ea8b5ffc9)

**Comment by cagnulein** [comment]
i suggest to start swapping the device to check if it's one it's better to
another to get and send metrics to mywhoosh.
also don't you see any wahoo virtual device on mywhoosh?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 8 set 2024 alle 20:53 dadude2607 ***@***.***>
ha scritto:

**Reply by dadude2607** [reply]
If I understand your answer correctly, then I should use a different device for qdomyos-zwift and mywoosh. That is the case. I use mywhoosh on my phone because it has enough power. The tablet runs qdomyos-zwift. I don't see a wahoo virtual device. The wahoo virtual device uses wlan, is that correct? Is mywhoosh capable of wlan devices? Is there any setting beside the "wahoo direct connect" I have to enable to use the wahoo virtual device? Do I need to turn of bluetooth virtual device?

**Reply by cagnulein** [reply]
What I suggest is to swap the devices that you are using. Maybe one device is better to another on the Bluetooth side

Also yes wahoo virtual device is using wlan and mywhoosh it's capable of! Enable the dircon setting in the experimental settings and the mywhoosh compatibility setting in the experimental settings as well.

Both devices must be connected to the same wifi network and wifi channel

Let me know

**Comment by dadude2607** [comment]
According to my research, direct connect is only available for PC. The option to activate direct connect is greyed out on Android. Since I don't have any other device to run qdomyos on, I'm going to get a raspberry pi ready for testing. I'll give you feedback

**Reply by cagnulein** [reply]
are you sure? i used mywhoosh with wahoo connection on my ipad

**Reply by dadude2607** [reply]
Not for Sure, but I found this information on a website and on my device it is grayed out. When starting mywhoosh on PC I can activate the direct connect. I will try to get another device and try bluetooth or will move my bike next to my gaming PC and try direct connect.

![Screenshot_20240909_181440_MyWhoosh](https://github.com/user-attachments/assets/e71fa2f1-928c-4d8e-a6ae-ba3262923897)

**Reply by dadude2607** [reply]
Some news....

I asked the guys from mywhoosh. Direct connect is not supported on Android.

But I was able to get an older Samsung Galaxy s10. The device shows up in mywhoosh as "Samsung Galaxy S10" without any special numbers or letters after that, like my Lenovo tablet does.

Did a 1 hour ride without any disconnect. So this seems to work. Maybe I could factory reset my Lenovo, but I am lazy 🤣 so I will use this device for future Workouts.

Thanks and regards
Andre

**Comment by cagnulein** [comment]
Glad to hear it!


Il giorno mer 11 set 2024 alle 21:51 dadude2607 ***@***.***>
ha scritto:
